### ISSUE TITLE
Comments appear differently in FireFox and Chrome (in Windows 10)

### ISSUE THREAD

**Issue by ToonTalk** [jmoenig/Snap--Build-Your-Own-Blocks#2146]
Here is a bit of the project in Chrome 67:

![image](https://user-images.githubusercontent.com/2138696/42584611-7facfc28-852b-11e8-88e2-c66b1a625fd9.png)

And here's the same bit in FireFox 61:

![image](https://user-images.githubusercontent.com/2138696/42584466-269c2d16-852b-11e8-837d-6cae8e818b93.png)

Here's the project:
[eCraft2Learn.zip](https://github.com/jmoenig/Snap--Build-Your-Own-Blocks/files/2185279/eCraft2Learn.zip)

**Comment by jmoenig** [comment]
Yeah, browsers handle font requests differently. Basically we don't know which font we'll be getting when we request one. Plus, users can even manually override font sizes and styles in different browsers and we can't do anything about it.

**Comment by ToonTalk** [comment]
Not sure how much it would help but you can find out the dimensions of the text. In canvas there is https://developer.mozilla.org/en-US/docs/Web/API/CanvasRenderingContext2D/measureText

I don't use the browser's canvas but it seems one can control the fonts: https://developer.mozilla.org/en-US/docs/Web/API/CanvasRenderingContext2D/font

**Comment by jmoenig** [comment]
Oh sure, we *do* totally measure text, that's how we layout the blocks and everything, and there is *some* control over fonts, but not much, plus it can be overridden by other settings. Since fonts are handled by the underlying system (i.e. browser) it's somewhat device dependent, as this example show.

**Comment by DarDoro** [comment]
There is another problem with measureText. Sometimes text-length of the entire string is less then  sum of text-length of every single character. It breaks some mouse cursor to character position calculation in Morphic.

As to font rendering there are post about posible "fix" :
https://www.reddit.com/r/firefox/comments/79nlx8/how_do_i_make_firefox_quantums_fonts_look_like/

https://www.reddit.com/r/firefox/comments/4ggqzo/psa_fix_firefox_fonts_on_windows/

**Comment by jmoenig** [comment]
Oh, that's interesting, @DarDoro ! So, maybe we could / should change the StringMorph's ```slotPosition``` method to measure sub-strings instead of adding up character widths.

**Comment by jmoenig** [comment]
Ah, I *knew* it! I missed something important... Let me revisit... Thanks!

**Comment by cycomachead** [comment]
> but not much, plus it can be overridden by other settings. Since fonts are handled by the underlying system (i.e. browser) it's somewhat device dependent, as this example shows.

We totally have control over this... but it's a bit pain. We could default to using a font which is consistent on all platforms, or very likely to be so -- like Arial. Or we could also ship a font file with with Snap! and use that as the basis for the font.
